### ISSUE TITLE
Cannot log the keys from an app on android device, Galaxy A54 5G

### ISSUE THREAD

**Issue by tomy1989** [fkie-cad/friTap#15]
I am trying to use the friTap.py tool for monitoring SSL/TLS traffic on my Android device. I've executed the script with the following command:

`./friTap.py -m -k keys.log -v -s com.google.android.youtube`

Upon running the script, I observed the following output:

```
Start logging
Press Ctrl+C to stop logging
spawning com.google.android.youtube
[*] Running Script on Android
[*] libssl.so found & will be hooked on Android!
[*] Android dynamic loader hooked.
[*] Logging keylog file to keys.log
[*] Blocking provider registration of Ssl_Guard
```

### However, I am facing two main issues:

1.     No Logging: Even though the script indicates that logging has started, I am not seeing any entries in the keys.log file.
2.     Blocking Ssl_Guard: The output mentions "Blocking provider registration of Ssl_Guard," but there is no additional information or log entries related to this action.

Could you please assist me in understanding what might be going wrong, and how to resolve these issues?

**Comment by monkeywave** [comment]
Hi,

thx for opening this issue. In order to reproduce the error we have some further questions.

- On what Android version are you running fritap?
- Can you provide the output of `./friTap.py -m -k keys.log -v -do -s com.google.android.youtube`

**Comment by tomy1989** [comment]
![image](https://github.com/fkie-cad/friTap/assets/21266504/2bd7538d-418e-401d-9ff3-281ce1a3b124)
using galaxy a54 5g with android version 13
device rooted using magisk

**Comment by Myshtea** [comment]
I have the same problem on an emulator "rooted" through magisk so I can confirm that this happens. Same outputs as @tomy1989 
Mine is configured like this :
  Name: PIXEL6_31_PLAYSTORE
  Device: pixel_6 (Google)
  Target: Google Play (Google Inc.)
          Based on: Android 12.0 ("S") Tag/ABI: google_apis_playstore/x86_64
  Sdcard: 512 MB

**Comment by tomy1989** [comment]
is there any update on this? :(

**Comment by monkeywave** [comment]
I'm still not sure where this problem is coming from. Does this only happen when running the Youtube app or also with other apps?

How does your output looks like when enabling spawn gating:
`python3 ./friTap.py -m -k keys.log -v --enable_spawn_gating -s com.google.android.youtube`

In my case frida gets a timeout.

**Comment by 3052** [comment]
YouTube Android uses QUIC, so unless you tool supports that you wont see the traffic.

as a workaround you can disable QUIC. you should be able to do this with Linux and Windows by blocking outbound UDP port 443. if you do this then it defaults back to HTTP/2

**Comment by monkeywave** [comment]
Should be resolved in the latest friTap version (version 1.2.1.0). 
Feel free to reopen it if the issue still occurs.

All the best

Daniel
